client/src/main/java/io/pravega/client/admin/impl/ReaderGroupManagerImpl.java

```diff
@@ -94,24 +95,43 @@ public void createReaderGroup(String groupName, ReaderGroupConfig config) {
         StateSynchronizer<ReaderGroupState> synchronizer = clientFactory.createStateSynchronizer(NameUtils.getStreamForReaderGroup(groupName),
                                               new ReaderGroupStateUpdatesSerializer(), new ReaderGroupStateInitSerializer(), SynchronizerConfig.builder().build());
         Map<SegmentWithRange, Long> segments = ReaderGroupImpl.getSegmentsForStreams(controller, config);
+
         synchronizer.initialize(new ReaderGroupState.ReaderGroupStateInit(config, segments, getEndSegmentsForStreams(config)));
+        @Cleanup
+        ReaderGroupImpl groupImpl = (ReaderGroupImpl) getReaderGroup(groupName);
+        val state = groupImpl.getState();
+        if (state.getConfigState() == ReaderGroupState.ConfigState.INITIALIZING && state.getConfig().equals(config)) {
+            groupImpl.doInit(state);
+        }
     }
 
     @Override
     public void deleteReaderGroup(String groupName) {
-        getAndHandleExceptions(controller.sealStream(scope, getStreamForReaderGroup(groupName))
-                                         .thenCompose(b -> controller.deleteStream(scope,
-                                                                                   getStreamForReaderGroup(groupName)))
-                                         .exceptionally(e -> {
-                                             if (e instanceof InvalidStreamException) {
-                                                 return null;
-                                             } else {
-                                                 log.warn("Failed to delete stream", e);
-                                             }
-                                             throw Exceptions.sneakyThrow(e);
-                                         }),
-                               RuntimeException::new);
-        
+        @Cleanup
+        ReaderGroupImpl groupImpl = (ReaderGroupImpl) getReaderGroup(groupName);
+
+        while (true) {
```

Comment:
is it possible to move this whole while(true) loop inside readergroupImpl ?
basically this method should do the following:
1. call delete on RGImpl which should update the RG state to follow the business logic here to modify the state to deleted.
2. then seal the RG stream and then delete it (the existing code).

Reply:
I've restructured the code as explained.